api/v1beta1/nodemaintenance_webhook.go

```diff
@@ -172,39 +174,19 @@ func (v *NodeMaintenanceValidator) validateControlPlaneQuorum(nodeName string) e
 		// not a control-plane node, nothing to do
 		return nil
 	}
-
-	// check the etcd-quorum-guard PodDisruptionBudget if we can drain a control-plane node
-	disruptionsAllowed := int32(-1)
-	for _, pdbName := range []string{EtcdQuorumPDBNewName, EtcdQuorumPDBOldName} {
-		var pdb policyv1.PodDisruptionBudget
-		key := types.NamespacedName{
-			Namespace: EtcdQuorumPDBNamespace,
-			Name:      pdbName,
-		}
-		if err := v.client.Get(context.TODO(), key, &pdb); err != nil {
-			if apierrors.IsNotFound(err) {
-				// try next one
-				continue
-			}
-			return fmt.Errorf("could not get the etcd quorum guard PDB for master/control-plane quorum validation, please try again: %v", err)
-		}
-		disruptionsAllowed = pdb.Status.DisruptionsAllowed
-		break
-	}
-	if disruptionsAllowed == -1 {
-		// TODO do we need a fallback for k8s clusters?
-		nodemaintenancelog.Info("etcd quorum guard PDB hasn't been found. Skipping master/control-plane quorum validation.")
-		return nil
+	canDisrupt, err := etcd.IsEtcdDisruptionAllowed(context.Background(), v.client, nodemaintenancelog, node)
```

Comment:
You are inventing new names again, which are less clear that what we already have (in the function name) 🤷🏼‍♂️
What's the advantage of "can" instead of "isAllowed"?
"Can" not only means if something is allowed, but it can (pun intended) also mean if you are able to do something. We are always able to disrupt etcd. But we should't always do that 😉

Reply:
also: you need to check if we are on OCP / OKD. On k8s we don't have the etcd PDB / guards, and the check returns false in that case, which stops maintenance. In the old version maintenance is allowed.